packages/components/src/Select/Select.rebuilt.test.tsx

```diff
@@ -0,0 +1,290 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Option } from "./Option";
+import { SelectRebuilt } from "./Select.rebuilt";
+
+describe("SelectRebuilt", () => {
+  it("renders a SelectRebuilt with no options", () => {
+    const { container } = render(<SelectRebuilt version={2} />);
+    expect(container).toMatchSnapshot();
+  });
+
+  it("renders a SelectRebuilt with one option", () => {
+    const { container } = render(
+      <SelectRebuilt version={2}>
+        <Option>Foo</Option>
+      </SelectRebuilt>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it("renders a SelectRebuilt with many options", () => {
+    const { container } = render(
+      <SelectRebuilt version={2}>
+        <Option>Foo</Option>
+        <Option>Bar</Option>
+        <Option>Baz</Option>
+        <Option>Quux</Option>
+      </SelectRebuilt>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it("renders correctly as small", () => {
+    const { container } = render(
+      <SelectRebuilt version={2} size="small">
+        <Option>Foo</Option>
+      </SelectRebuilt>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it("renders correctly as large", () => {
+    const { container } = render(
+      <SelectRebuilt version={2} size="large">
+        <Option>Foo</Option>
+      </SelectRebuilt>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it("renders correctly when disabled", () => {
+    const { container } = render(
+      <SelectRebuilt version={2} disabled>
+        <Option>Foo</Option>
+      </SelectRebuilt>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it("renders correctly when invalid", () => {
+    const { container } = render(
+      <SelectRebuilt version={2} invalid>
+        <Option>Foo</Option>
+      </SelectRebuilt>,
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it("renders the value when set", () => {
+    render(
+      <SelectRebuilt version={2} value="bar">
+        <Option value="foo">Foo</Option>
+        <Option value="bar">Bar</Option>
+      </SelectRebuilt>,
+    );
+
+    const selectElement = screen.getByRole("combobox");
+    expect(selectElement).toHaveValue("bar");
+  });
+
+  it("should set the selected value when given 'value'", () => {
+    const expectedValue = "baz";
+
+    render(
+      <SelectRebuilt version={2} value={expectedValue}>
+        <Option value="foo">Foo</Option>
+        <Option value="bar">Bar</Option>
+        <Option value="baz">Baz</Option>
+      </SelectRebuilt>,
+    );
+
+    const selectElement = screen.getByRole("combobox");
+
+    // Verify the value is correctly set
+    expect(selectElement).toHaveValue(expectedValue);
+
+    // Verify the correct option is selected
+    const select = selectElement as HTMLSelectElement;
+    expect(select.options[select.selectedIndex].text).toBe("Baz");
```

Comment:
You could do the following but it isn't blocking

```suggestion
     expect(select.options[select.selectedIndex]).toHaveTextContent("Baz");
```